proposals/carvel/002-artifact-signatures/README.md

```diff
@@ -0,0 +1,279 @@
+cosign sign-blob \
+  imgpkg-darwin-arm64 \
+  --output-certificate imgpkg-darwin-arm64.pem \
+  --output-signature imgpkg-darwin-arm64.sig
+```
+
+The command will output certificate and signature in the specified files, which must be distributed together with the binary artifact. Users can verify the signature for a given binary artifact using the Cosing CLI. The snippet assumes that the signing process has been performed as part of a pipeline based on GitHub Actions using the Workflow reference as the identity.
+
+```shell script
+cosign verify-blob \
+  --cert imgpkg-darwin-arm64.pem \
+  --signature imgpkg-darwin-arm64.sig \
+  --certificate-identity-regexp=https://github.com/carvel-dev \
+  --certificate-oidc-issuer=https://token.actions.githubusercontent.com \
+  imgpkg-darwin-arm64
+```
+
+If the signature is valid for both the binary artifact and the `checksums.txt` file, users can proceed verifying the `SHA256` sums match the downloaded binary as they would usually do. For example, using the `shasum` utility.
+
+```shell script
+shasum -a 256 imgpkg-darwin-arm64
+```
+
+For more information about signing blobs with Cosign, refer to the [official documentation](https://docs.sigstore.dev/cosign/signing_with_blobs/).
+
+##### Binary artifacts to sign
+
+The proposal is for signing all the artifacts included in the following Carvel projects.
+
+**imgpkg**:
+
+* `imgpkg-darwin-amd64`
+* `imgpkg-linux-amd64`
+* `imgpkg-windows-amd64.exe`
+* `imgpkg-darwin-arm64`
+* `imgpkg-linux-arm64`
+* `checksums.txt`
+
+**kapp**:
+
+* `kapp-darwin-amd64`
+* `kapp-linux-amd64`
+* `kapp-windows-amd64.exe`
+* `kapp-darwin-arm64`
+* `kapp-linux-arm64`
+* `checksums.txt`
+
+**kapp-controller**:
+
+* `kctrl-darwin-amd64`
+* `kctrl-linux-amd64`
+* `kctrl-windows-amd64.exe`
+* `kctrl-darwin-arm64`
+* `kctrl-linux-arm64`
+* `checksums.txt`
+
+**kbld**:
+
+* `kbld-darwin-amd64`
+* `kbld-linux-amd64`
+* `kbld-windows-amd64.exe`
+* `kbld-darwin-arm64`
+* `kbld-linux-arm64`
+* `kbld-windows-arm64.exe`
+* `checksums.txt`
+
+**vendir**:
+
+* `vendir-darwin-amd64`
+* `vendir-linux-amd64`
+* `vendir-windows-amd64.exe`
+* `vendir-darwin-arm64`
+* `vendir-linux-arm64`
+* `checksums.txt`
+
+**ytt**:
+
+* `ytt-darwin-amd64`
+* `ytt-linux-amd64`
+* `ytt-windows-amd64.exe`
+* `ytt-darwin-arm64`
+* `ytt-windows-arm64.exe`
+* `ytt-linux-arm64`
+* `checksums.txt`
+
+#### Using the Sigstore GitHub Action
+
+The Sigstore project provides an official Action to set up Cosign in a pipeline based on GitHub Actions: [sigstore/cosign-installer](https://github.com/sigstore/cosign-installer).
+
+Example using a container image:
+
+```yaml
+name: Commit Stage
+on: push
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+      packages: write
+      id-token: write # Required by Cosign to authenticate with GitHub via OIDC
+    steps:
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@v3
+      - name: Sign image
```

Comment:
We can guess how signing binaries from the above context.
I think it would be helpful to have an example of signing a binary over here as well.
I believe it would be as simple as using the other command

Reply:
Good idea, I'll add that

Reply:
I've added an example